mat/pool.go

```diff
@@ -63,6 +63,9 @@ var (
 
 	// poolCmplx is the CDense equivalent of pool.
 	poolCmplx [63]sync.Pool
+
+	// TODO(notmgsk): wtf
```

Comment:
The magic here is that there are a collection of exponentially sized sources of temporary work slices. This means that when we need a temporary slice, we ask for one and we know that it will be not more that 2*n-1 the length of what we need. We do this to avoid allocation of temporary slices.